src/Unleash/ClientFactory/UnleashClientFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Unleash.ClientFactory
+{
+    public class UnleashClientFactory : IUnleashClientFactory
+    {
+        private UnleashSettings settings { get; }
+        private CancellationTokenSource cancellationTokenSource { get; } = new CancellationTokenSource();
+
+
+        public UnleashClientFactory(UnleashSettings settings)
+        {
+            this.settings = settings;
+        }
+
+
+        public async Task<IUnleash> Generate(bool SynchronousInitialization = false)
```

Comment:
Is this parameter needed? Perhaps this should be the default behavior

Reply:
Let maintain the same behavior using new and factory.